### ISSUE TITLE
"Do not disturb me" feature :)

### ISSUE THREAD

**Issue by tio-trom** [friendica/friendica#11848]
Ok this may not be on anyone's priority list but I'll try to condense my argument:

"social networks" are usually "ad networks" (facebook, twitter, etc.) - their motivation is not the well being of our minds, but to consume as much as possible. More comments, more likes, more shares, more more. The Fediverse tries to do the opposite and provide a cozy place for our brains. But it may have copied the other networks too much and lacks some features that I think are crucial.

**Do not disturb.**

You post all sort of things on Friendica but at times you need some piece of mind and do not want anyone to bother you by engaging/commenting on any of your posts. It is a normal human need to have the ability to feel like you are in your own house and no one can bother you. Thus, I was thinking of such a unique and damn-useful feature in Friendica. To have a toggle that disables the comments or any interactions to all of your posts. How feasible would that be? Toggle it on again and is back to normal.

I have not see this feature on any such platforms so I think Friendica would stand out as a Social Network that cares for the well-being of people if it had this feature. 

And if this is too difficult to implement, at least if one could do that for individual posts. Say I make a post and I do not want to enable any comments for it. Even FB has this sort of option. Could be very useful for many's mental health.

In any case, if anyone is interested in this we can discuss more about the details and implications of such a feature.

**Comment by MrPetovan** [comment]
I don’t think it’s feasible because of the federated nature of Friendica. Even if we could pass a DND flag with the post when we transmit it, non-Friendica platforms wouldn’t honor it.

We could just refuse remote interactions on such posts, but it would give you a false view of the conversation happening below your post. It’s already bad enough with blocking that is necessary but has only a limited scope in federated social media, I’m not sure we should venture further in that vein.

On the other hand, we could mute notifications as long as DND is enabled, I know it isn’t as radical but it would limit the interruptions.

**Comment by tobiasd** [comment]
> On the other hand, we could mute notifications as long as DND is enabled, I know it isn’t as radical but it would limit the interruptions.

That sounds like a nice idea. Could this be implemented in a way, that clients
wont get new notifications via API during the DND time?

**Comment by tio-trom** [comment]
I see. But it is all about what you see. So if say I make a post. In the fediverse the post will be spread across multiple instances. Now some people comment on it, and I see their comments and interactions. I then enable DND and any comment posted won't reach me, no notifications, no comment visible on my post. Yes these comments may appear on other networks but can't you push a "delete comment from my post" to other instances? How does it work when I edit my post and the edit gets pushed to the other networks?

Basically if others comment on my post via other instances, their comments will get deleted on my end, and thus not visible.

Am trying to understand how this may work. But am surprised there is no push for "delete this comment from my post" and that will apply across the fediverse. This is very useful for spam, bots, harassment, blocking others, etc.

**Comment by MrPetovan** [comment]
So, a “Delete *this* comment from my post” could work, provided remote platform honor it. This is not guaranteed because you’re deleting someone else’s content so authorship checks would fail.

But “Delete an unspecified number of comments under this post of mine” wouldn’t work at all, it’s too vague with no reference to the specific content (for a good reason, your server wouldn’t know about it since it rejected it), and this request would need to be repeated regularly whether there’s offending content or not.

**Comment by MrPetovan** [comment]
> On the other hand, we could mute notifications as long as DND is enabled, I know it isn’t as radical but it would limit the interruptions.
> That sounds like a nice idea. Could this be implemented in a way, that clients wont get new notifications via API during the DND time?

I don’t know how notifications work with apps, but the concept of DND would be that notifications are created with the “seen” flag already set, which shouldn’t trigger any visible notifications either on the web frontend or in the apps.

**Comment by tio-trom** [comment]
Thank you for detailing all of this for me. It is good to know. I think a sort of DND would be useful even if it is far from what I first imagined, at least something to prevent any notifications from showing. That would be a great start I think. "notifications are created with the “seen” flag already set," - sounds reasonable to me. Perhaps they should also be hidden from the notification area.

> So, a “Delete this comment from my post” could work, provided remote platform honor it. This is not guaranteed because you’re deleting someone else’s content so authorship checks would fail.

I understand. It is still a very useful implementation especially since it can be used for blocked contacts. Right now, as far as I am aware, if someone trolls me, spams, or whatever my posts, if I block the account I simply won't see hos ongoing comments. But my friends can still see them. This is really bothersome really. I can't mute these accounts even if I wanted to. But if we can auto-delete their comments so that no one sees them...that could be really useful. Activitypub really needs better tools to deal with such situations. The other networks should honor such things.
